newsfragments/43915.significant.rst

```diff
@@ -0,0 +1,4 @@
+Configuration ``[core] strict_dataset_uri_validation`` is removed
```

Comment:
This is not a typo; the configuration key uses _dataset_ on Airflow 2, which users will be upgrading from.